File: q1877/q1877.java
class Solution {
    public int minPairSum(int[] nums) {
        Arrays.sort(nums);
        int minSum = Integer.MIN_VALUE;
        int l = 0;
        int r = nums.length - 1;
        while (l < r) {
            minSum = Math.max(minSum, nums[l] + nums[r]);
            l++;
            r--;
        }
        return minSum;
    }
}

File: q1877/q1877.cpp
class Solution
{
public:
    int minPairSum(vector<int> &nums)
    {
        sort(nums.begin(), nums.end());
        int minSum = INT_MIN;
        int l = 0;
        int r = nums.size() - 1;
        while (l < r)
        {
            minSum = max(nums[l] + nums[r], minSum);
            l++;
            r--;
        }
        return minSum;
    }
};